### PR TITLE
fix(*): refactor reusable styles [KHCP-7735]

### PR DESCRIPTION
**Pull request by portikM**
# Summary

Refactor reusable styles into components and mixins.

Addresses: https://konghq.atlassian.net/browse/KHCP-7735

## PR Checklist

* [ ] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/kongponents#committing-changes).
* [ ] **Functional:** all changes do not break existing APIs and if so, bump major version.
* [ ] **Tests pass:** check the output of yarn test
* [ ] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
* [ ] **Framework style:** abides by the essential rules in Vue's style guide
* [ ] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs, debugger), or leftover comments
* [ ] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: src/components/KInputSwitch/KInputSwitch.vue [hunk 1]
```diff
@@ -160,13 +160,19 @@ export default {
 
 <style lang="scss" scoped>
 @import '@/styles/variables';
-@import '@/styles/functions';
+@import "@/styles/tmp-variables";
```
Comment (adamdehaven) [inline src/components/KInputSwitch/KInputSwitch.vue:163]: fix: can you standardize to single quotes on all of the imports?

File: src/components/KInputSwitch/KInputSwitch.vue [hunk 2]
```diff
@@ -160,13 +160,19 @@ export default {
...
 
-.k-switch,
-.k-input-switch {
+$width: 44px;
+$height: 24px;
+$transition: $tmp-animation-timing-2 linear;
```
Comment (adamdehaven) [inline src/components/KInputSwitch/KInputSwitch.vue:167]: suggestion: can you name these a little more specifically if only used here? e.g. `$k-input-switch-width` etc.

File: docs/guide/styles/standalone-usage.md [hunk 3]
```diff
@@ -41,7 +41,7 @@ You can even **scope** the styles and/or variables to a container class to preve
 @use "sass:meta";
 
 // Import Kongponents Sass variables (path may vary)
-@import "@kong/kongponents/dist/_variables.scss";
+@import '@kong/kongponents/dist/_variables.scss';
```
Comment (adamdehaven) [inline docs/guide/styles/standalone-usage.md:44]: I think this whole section of the docs needs to be removed now, right?

File: src/components/KCheckbox/KCheckbox.vue [hunk 4]
```diff
@@ -122,42 +122,46 @@ export default {
...
+      color: var(--KCheckboxDisabledChecked, var(--grey-400, var(--kui-color-text-disabled, $kui-color-text-disabled)));
+    }
   }
 }
 </style>
 
 <style lang="scss">
 .k-checkbox-description {
-  font-weight: var(--kui-font-weight-semibold, $kui-font-weight-semibold);;
+  font-weight: var(--kui-font-weight-semibold, $kui-font-weight-semibold);
```
Comment (adamdehaven) [inline src/components/KCheckbox/KCheckbox.vue:159]: question: why isn't this style block scoped? (when we reskin Kongponents, **all styles must be scoped** to the component they live in -- going to make a ticket for this)

File: src/components/KLabel/KLabel.vue [hunk 5]
```diff
@@ -87,5 +105,9 @@ const hasTooltip = computed((): boolean => !!(props.info || props.help || slots.
       }
     }
   }
+
+  .kong-icon {
+    margin-left: var(--spacing-xxs, var(--kui-space-20, $kui-space-20));
+  }
```
Comment (adamdehaven) [inline src/components/KLabel/KLabel.vue:111]: does this need the `:deep` selector?

File: src/styles/_mixins.scss [hunk 6]
```diff
@@ -36,7 +38,7 @@
 @mixin textarea-default {
   background-color: var(--KInputBackground, var(--white, #ffffff));
   box-shadow: none !important;
-  outline: 1px solid var(--KInputBorder, var(--grey-300, #e7e7ec)) !important;
+  // outline: 1px solid var(--KInputBorder, var(--grey-300, #e7e7ec)) !important;
```
Comment (adamdehaven) [inline src/styles/_mixins.scss:41]: question: why is this line commented out?

**Comment by adamdehaven** [comment]
The radio styles look incorrect -- they are being shown as checkboxes 
image

**Comment by adamdehaven** [comment]
KInputSwitch looks the same regardless of the toggle state
image

File: docs/guide/styles/standalone-usage.md (hunk 3 above)
Comment (portikM) [reply docs/guide/styles/standalone-usage.md]: Yup pretty much. I was thinking about it and I suggest we create one story for docs and add a sub-ticket to go through all non-component related docs pages to make sure docs related to legacy functionality are removed (like this page). Writing migration guide could be another sub-ticket in that story.

Comment (adamdehaven) [reply docs/guide/styles/standalone-usage.md]: I already removed it - this is resolved; I'll add some docs tickets

File: src/components/KLabel/KLabel.vue (hunk 5 above)
Comment (portikM) [reply src/components/KLabel/KLabel.vue:111]: I would think so but it works without it so I left it as is.

File: src/components/KCheckbox/KCheckbox.vue (hunk 4 above)
Comment (portikM) [reply src/components/KCheckbox/KCheckbox.vue:159]: Good point, moving those styles to scoped block.

File: src/styles/_mixins.scss (hunk 6 above)
Comment (portikM) [reply src/styles/_mixins.scss]: Ended up removing this because it's now overwritten by another mixin.

**Comment by portikM** [comment]
> The radio styles look incorrect -- they are being shown as checkboxes image

Addressed

**Comment by portikM** [comment]
> KInputSwitch looks the same regardless of the toggle state image
> 
> image

Fixed

**Comment by adamdehaven** [comment]
Noticed a couple more things (some of these appear to be broken on `main` as well, but should be fixed now in this PR):

1. `KRadio`: Tooltip placement is broken
    image
2. `KRadio` and `KCheckbox`: The `Theming` examples are broken
    image
3. `KCheckbox`: the `HTML attributes` do not match - the label doesn't look "disabled"
    image
4. `KCheckbox`: The "privacy policy" link doesn't have the proper spacing
    image

**Comment by portikM** [comment]
> Noticed a couple more things (some of these appear to be broken on `main` as well, but should be fixed now in this PR):
> 
> 1. `KRadio`: Tooltip placement is broken
>    image
> 2. `KRadio` and `KCheckbox`: The `Theming` examples are broken
>    image
> 3. `KCheckbox`: the `HTML attributes` do not match - the label doesn't look "disabled"
>    image
> 4. `KCheckbox`: The "privacy policy" link doesn't have the proper spacing
>    image

All issues fixed

File: docs/components/checkbox.md [hunk 7]
```diff
@@ -131,7 +131,7 @@ Anything passed in to the default slot will replace the label prop text
     </div>
     <div>
       <KCheckbox v-model="slots2">
-        I agree to the <a href="#slots">privacy policy</a>.
+        I agree to the&nbsp;<a href="#slots">privacy policy</a>.
```
Comment (adamdehaven) [inline docs/components/checkbox.md:134]: The actual issue here is that the label is `inline-flex` - should it be?

Comment (portikM) [reply docs/components/checkbox.md:134]: Yes, it was defined in `_labels.scss` styles file then moved to component style block

File: src/styles/forms/_labels.scss [hunk 8]
```diff
@@ -1,40 +0,0 @@
-// some properties in this file are using kui design tokens because Kongponent styles are relying on matching values
-.k-input-label {
-  color: var(--KInputLabelColor, var(--black-85));
-  display: inline-flex;
```
Comment (portikM) [inline src/styles/forms/_labels.scss:4]: @adamdehaven here

**Review by adamdehaven — APPROVED** [review]

**Pull request merged**